legacy-sdk/whirlpool/tests/integration/v1-ix/open_position_with_token_extensions.test.ts

```diff
@@ -512,6 +517,199 @@ describe("open_position_with_token_extensions", () => {
+        params.positionPda.publicKey,
+      )) as PositionData;
+      assert.strictEqual(position.tickLowerIndex, expectedLower);
+      assert.strictEqual(position.tickUpperIndex, upper);
+    });
+
+    it("snaps upper to floor(current) when upper sentinel is used (token-2022)", async () => {
+      const whirlpool = await fetcher.getPool(
+        whirlpoolPda.publicKey,
+        IGNORE_CACHE,
+      );
+      if (!whirlpool) throw new Error("whirlpool not found");
+      const spacing = whirlpool.tickSpacing;
+      const curr = whirlpool.tickCurrentIndex;
+      const expectedUpper = snapTickDown(curr, spacing);
+      const lower = expectedUpper - spacing;
+
+      const { params, mint } =
+        await generateDefaultOpenPositionWithTokenExtensionsParams(
+          ctx,
+          whirlpoolPda.publicKey,
+          true,
+          lower,
+          SENTINEL_MAX,
+          ctx.wallet.publicKey,
+        );
+      await toTx(
+        ctx,
+        WhirlpoolIx.openPositionWithTokenExtensionsIx(ctx.program, params),
+      )
+        .addSigner(mint)
+        .buildAndExecute();
+
+      const position = (await fetcher.getPosition(
+        params.positionPda.publicKey,
+      )) as PositionData;
+      assert.strictEqual(position.tickLowerIndex, lower);
+      assert.strictEqual(position.tickUpperIndex, expectedUpper);
+    });
+
+    it("fails if both sentinels are used (token-2022)", async () => {
+      const { params, mint } =
+        await generateDefaultOpenPositionWithTokenExtensionsParams(
+          ctx,
+          whirlpoolPda.publicKey,
+          true,
+          SENTINEL_MIN,
+          SENTINEL_MAX,
+          ctx.wallet.publicKey,
+        );
+      await assert.rejects(
+        toTx(
+          ctx,
+          WhirlpoolIx.openPositionWithTokenExtensionsIx(ctx.program, params),
+        )
+          .addSigner(mint)
+          .buildAndExecute(),
+        /0x177a/, // InvalidTickIndex
+      );
+    });
+
+    it("fails if snapped range collapses (lower >= upper) (token-2022)", async () => {
+      const whirlpool = await fetcher.getPool(
+        whirlpoolPda.publicKey,
+        IGNORE_CACHE,
+      );
+      if (!whirlpool) throw new Error("whirlpool not found");
+      const spacing = whirlpool.tickSpacing;
+      const curr = whirlpool.tickCurrentIndex;
+      const expectedLower = snapTickUp(curr, spacing);
+      const upperEqualsLower = expectedLower;
+
+      const { params, mint } =
+        await generateDefaultOpenPositionWithTokenExtensionsParams(
+          ctx,
+          whirlpoolPda.publicKey,
+          true,
+          SENTINEL_MIN,
+          upperEqualsLower,
+          ctx.wallet.publicKey,
+        );
+      await assert.rejects(
+        toTx(
+          ctx,
+          WhirlpoolIx.openPositionWithTokenExtensionsIx(ctx.program, params),
+        )
+          .addSigner(mint)
+          .buildAndExecute(),
+        /0x177a/, // InvalidTickIndex
+      );
+    });
+  });
+
```

Comment:
Could we add test to check event emission ? (for 4 instructions)
(Other events have such kind of test case)